src/i18n.ts

```diff
@@ -0,0 +1,280 @@
+    } else {
+      this.config.fluentBundleOptions = {
+        ...this.config.fluentBundleOptions,
+        functions: { CTX },
+      };
+    }
+
+    this.fluent = new Fluent(this.config.fluentOptions);
+
+    if (config.directory) {
+      this.loadLocalesDir(config.directory);
+    }
+  }
+
+  /**
+   * Loads locales from the specified directory and registers them in the Fluent instance.
+   * @param directory Path to the directory to look for the translation files.
+   */
+  loadLocalesDir(directory: string): void {
+    for (const file of readLocalesDir(directory)) {
+      const path = resolve(directory, file);
+      const locale = file.substring(0, file.lastIndexOf("."));
+
+      this.loadLocale(locale, {
+        filePath: path,
+        bundleOptions: this.config.fluentBundleOptions,
+      });
+    }
+  }
+
+  /**
+   * Registers a locale in the Fluent instance based on the provided options.
+   * @param locale Locale ID
+   * @param options Options to specify the source and behavior of the translation
+   */
+  loadLocale(
+    locale: LocaleId,
+    options: {
+      filePath?: string;
+      source?: string;
+      isDefault?: boolean;
+      bundleOptions?: FluentBundleOptions;
+    },
+  ): void {
+    this.fluent.addTranslation({
+      locales: locale,
+      isDefault: locale === this.config.defaultLocale,
+      bundleOptions: this.config.fluentBundleOptions,
+      ...options,
+    });
+
+    this.locales.push(locale);
+  }
+
+  /** Gets an array of translations for a given key. */
+  t(key: string): Array<string>;
+  /** Gets a message by its key from the specified locale. */
+  t(locale: LocaleId, key: string, context?: TranslationContext): string;
+  t(
+    keyOrLocale: string | LocaleId,
+    key?: string,
+    context?: TranslationContext,
+  ): string | Array<string> {
+    if (keyOrLocale && key) {
+      return this.fluent.translate(keyOrLocale, key, context);
+    }
+
+    const translations = new Array<string>();
+    for (const locale of this.locales) {
+      const message = this.translate(locale, keyOrLocale, context);
+      translations.push(message);
+    }
+
+    return translations;
+  }
+
+  /** Gets a message by its key from the specified locale. */
+  translate(key: string): Array<string>;
+  translate(
+    locale: LocaleId,
+    key: string,
+    context?: TranslationContext,
+  ): string;
+  translate(
+    keyOrLocale: string,
+    key?: string,
+    context?: TranslationContext,
+  ): string | string[] {
+    if (keyOrLocale && key) {
+      return this.fluent.translate(keyOrLocale, key, context);
+    }
+
+    const translations = new Array<string>();
+    for (const locale of this.locales) {
+      const message = this.translate(locale, keyOrLocale, context);
+      translations.push(message);
+    }
+
+    return translations;
+  }
+
+  /** Returns a middleware to .use on the `Bot` instance. */
+  middleware(): Middleware<C & I18nContextFlavor> {
+    const fluent = this.fluent;
```

Comment:
But, you cannot access `this.fluent` in the middleware function we're returning. That is why I stored it as a variable instead of reading it from `this.fluent`.

Reply:
My bad, I have this thought while I was detaching the function... I shouldn't have written it in here :D